### ISSUE TITLE
About the sigmoid activation funcion

### ISSUE THREAD

**Issue by HarryHsing** [aharley/pips#20]
Hi, I cannot find the sigmoid activation layer in the code, but it is mentioned in the paper, as shown in the figure below. I wonder if it exist or not.

Thank you very much for your support!

![image](https://user-images.githubusercontent.com/36613867/222906352-5feeba54-e3f0-4821-a5c4-c8a4e473c8eb.png)

**Comment by ZHAOZHIHAO** [comment]
Hi, although I'm not the author, I may be able to answer your question. The sigmoid exists as F.binary_cross_entropy_with_logits = F.sigmoid + F.binary_cross_entropy, which seems a trick to avoid numerical unstable. A more popular and exactly same thing is CrossEntropyLoss, which combines LogSoftmax() and NLLLoss() so that you won't see softmax, as pointed out here https://stackoverflow.com/questions/55675345/should-i-use-softmax-as-output-when-using-cross-entropy-loss-in-pytorch

**Comment by HarryHsing** [comment]
> Hi, although I'm not the author, I may be able to answer your question. The sigmoid exists as F.binary_cross_entropy_with_logits = F.sigmoid + F.binary_cross_entropy, which seems a trick to avoid numerical unstable. A more popular and exactly same thing is CrossEntropyLoss, which combines LogSoftmax() and NLLLoss() so that you won't see softmax, as pointed out here https://stackoverflow.com/questions/55675345/should-i-use-softmax-as-output-when-using-cross-entropy-loss-in-pytorch

Thank you very much for your detailed reply! It really helps me a lot, thx!
